src/main/java/de/hpi/bpt/argos/parsing/EventTypeParserImpl.java

```diff
@@ -0,0 +1,221 @@
+		}
+
+		return instance;
+	}
+
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void loadEventTypes() {
+		String eventTypesDirectoryPath = Argos.getEventTypesDirectory();
+		if (eventTypesDirectoryPath.length() == 0) {
+			logger.info("event types directory not defined in properties");
+			return;
+		}
+
+		// since the path is delivered as URI, it is represented as a HTML string. Thus we need to replace %20 with file system spaces.
+		File eventTypesDirectory = new File(eventTypesDirectoryPath.replaceAll("%20", " "));
+
+		try {
+			for (File eventTypeFile : eventTypesDirectory.listFiles()) {
+				if (!eventTypeFile.getName().endsWith(".json")) {
+					continue;
+				}
+
+				loadDefaultEventType(eventTypeFile);
+			}
+
+		} catch (NullPointerException e) {
+			LoggerUtilImpl.getInstance().error(logger, "cannot find event types directory", e);
+		}
+	}
+
+	/**
+	 * This method loads the content of a specific file and parses this into an event type.
+	 * @param eventTypeFile - the file to load
+	 */
+	private void loadDefaultEventType(File eventTypeFile) {
+		try {
+			if (!wasModified(eventTypeFile)) {
+				logger.info(String.format("file '%1$s' was loaded already and therefore skipped", eventTypeFile.getName()));
+				return;
+			} else {
+				modify(eventTypeFile);
+			}
+
+			String fileContent = new String(Files.readAllBytes(Paths.get(eventTypeFile.toURI())), StandardCharsets.UTF_8);
+
+			JsonObject jsonEventType = jsonParser.parse(fileContent).getAsJsonObject();
+
+			if (jsonEventType.get(JSON_NAME).getAsString().isEmpty()) {
+				return;
+			}
+
+			List<EventType> eventTypes = PersistenceAdapterImpl.getInstance().getEventTypes();
+
+			for (EventType eventType : eventTypes) {
+				if (eventType.getName().equalsIgnoreCase(jsonEventType.get(JSON_NAME).getAsString())) {
+					return;
+				}
+			}
+
+			createEventType(jsonEventType);
+
+		} catch (Exception e) {
+			LoggerUtilImpl.getInstance().error(logger, String.format("cannot load event type from file '%1$s'", eventTypeFile.getName()), e);
+		}
+	}
+
+	/**
+	 * This method checks whether a file was modified since the last time it was loaded.
+	 * @param file - the file to check for modification
+	 * @return - true, if the file was modified and therefore should be loaded again
+	 */
+	private boolean wasModified(File file) {
+		DataFile dataFile = PersistenceAdapterImpl.getInstance().getDataFile(file.getAbsolutePath());
+
+		return dataFile == null || dataFile.getModificationTimestamp() != file.lastModified();
+	}
+
+	/**
+	 * This method updated the modificationTimestamp for a given file.
+	 * @param file - the file to be updated
+	 */
+	private void modify(File file) {
+		DataFile dataFile = PersistenceAdapterImpl.getInstance().getDataFile(file.getAbsolutePath());
+
+		if (dataFile == null) {
+			dataFile = new DataFileImpl();
+			dataFile.setPath(file.getAbsolutePath());
+		}
+
+		dataFile.setModificationTimestamp(file.lastModified());
```

Comment:
In my opinion it is better to save the actual modification timestamp instead of the time we last checked.